src/pages/K8s.vue

```diff
@@ -361,12 +361,15 @@ export default {
     },
     async handleDisableKubernetesCheckbox(value) {
       if (value !== this.settings.kubernetes.enabled) {
-        const confirmationMessage = [`${ value ? 'Enabling' : 'Disabling' } Kubernetes requires a restart. `,
-          '\n\nDo you want to proceed?'].join('');
+        const confirmationMessage = [
+          `${ value ? 'Enabling' : 'Disabling' } Kubernetes requires a restart. `,
+          '\n\nDo you want to proceed?'
+        ].join('');
 
         if (confirm(confirmationMessage)) {
           try {
             await ipcRenderer.invoke('settings-write', { kubernetes: { enabled: value } });
+            ipcRenderer.send('dashboard-close');
```

Comment:
Thinking about the CLI, the request to shutdown the dashboard should be sent from the code in `background.ts` that processes the `settings-write` event. Thinking along the lines of having UI event handlers do as little as possible, which in most cases should be working with widgets and sending events.

Reply:
I removed this and implemented some logic in the `settings-write` listener to check if kubernetes is disabled before closing the dashboard.